bodo/pandas/plan_optimizer.pyx

```diff
@@ -394,15 +395,24 @@ cdef class LogicalGetParquetRead(LogicalOperator):
     def __str__(self):
         return f"LogicalGetParquetRead({self.path})"
 
+
 cdef class LogicalGetSeriesRead(LogicalOperator):
     """Represents an already materialized BodoSeries."""
     def __cinit__(self, result_id):
         assert False & "Not implemented yet."
 
+
 cdef class LogicalGetDataframeRead(LogicalOperator):
     """Represents an already materialized BodoDataFrame."""
-    def __cinit__(self, result_id):
-        assert False & "Not implemented yet."
+    cdef readonly object arrow_schema
```

Comment:
The purpose of this data source was to represent a distributed materialization of a previous plan execution.  If you want a data source for a non-distributed incoming pandas dataframe then please create a new datasource class.

Reply:
Added a new class.